benchexec/tablegenerator/test_util.py

```diff
@@ -51,3 +52,49 @@ def test_split_string_at_suffix(self):
         self.assertEqualTextAndNumber("abc", "abc", "")
         self.assertEqualTextAndNumber("abc1abc", "abc1abc", "")
         self.assertEqualTextAndNumber("abc1abc1", "abc1abc", "1")
+
+    def test_print_decimal_roundtrip(self):
+        # These values should be printed exactly as in the input (with "+" removed)
+        test_values = [
+            "NaN",
+            "Inf",
+            "-Inf",
+            "+Inf",
+            "0",
+            "-0",
+            "+0",
+            "0.0",
+            "-0.0",
+            "0.00000000000000000000",
+            "0.00000000000000000001",
+            "0.00000000123450000000",
+            "0.1",
+            "0.10000000000000000000",
+            "0.99999999999999999999",
+            "1",
+            "-1",
+            "+1",
+            "1000000000000000000000",
+            "10000000000.0000000000",
+        ]
+        for value in test_values:
+            expected = value.lstrip("+")
+            self.assertEqual(expected, util.print_decimal(Decimal(value)))
+
+    def test_print_decimal_int(self):
+        # These values should be printed like Decimal prints them after quantizing
+        # to remove the exponent.
+        test_values = ["0e0", "-0e0", "0e20", "1e0", "1e20", "0e10"]
+        for value in test_values:
+            value = Decimal(value)
+            expected = str(value.quantize(1))
+            assert "e" not in expected
```

Comment:
[non-blocking minor] inconsistent usage of assertion, no message available

Reply:
Thanks for looking at this!

Yes, assertion messages are in general inconsistent in BenchExec, which is ok for me because we don't expect them to be shown to users. And if users do happen to see them, they need to report a bug and we need to debug this anyway, so it is just a minor hurdle then.